jwst/outlier_detection/outlier_detection.py

```diff
@@ -182,20 +183,14 @@ def do_detection(self):
         self.build_suffix(**self.outlierpars)
 
         pars = self.outlierpars
-        save_intermediate_results = pars['save_intermediate_results']
+
         if pars['resample_data']:
             # Start by creating resampled/mosaic images for
             # each group of exposures
             resamp = resample.ResampleData(self.input_models, single=True,
                                            blendheaders=False, **pars)
             drizzled_models = resamp.do_drizzle()
-            if save_intermediate_results:
-                for model in drizzled_models:
-                    log.info("Writing out resampled exposures...")
-                    model_output_path = self.make_output_path(
-                        basepath=model.meta.filename,
-                        suffix='outlier_i2d')
-                    model.save(model_output_path)
+
```

Comment:
I do not understand why `if save_intermediate_results: ...` block was removed. Is saving intermediate results no longer needed?

Reply:
It's my understanding that all intermediate results are now always saved to disk (instead of keeping them in memory only), hence there's no need for the optional param anymore. Not sure what happens if the new param `in_memory` is activated.

Reply:
This parameter actually controls the behavior of writing out all the final resampled products when they are created. Unfortunately, it was not practical to keep the rest of the **outlier_detection** step compatible with both in-memory and on-disk processing, so this parameter does not get used for the **outlier_detection** products (intermediate or final).   As such, the intermediate results are always written out to disk making the '**save_intermediate_results**' code duplicative during processing.